### PR TITLE
Add tests for build history and remove initiating of Client()

### PR DESCRIPTION
**Pull request by arjunsalyan**

### CODE REVIEW

**Comment by mojca** [comment]
I don't have any further suggestion for this (honestly I'm not sure about all the details), but the conflict needs to be resolved. If @umeshksingla doesn't have anything to add, feel free to merge.

File: app/ports/tests/test_buildhistory.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+class TestDependencies(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        Port.load(TEST_PORTINDEX_JSON)
+
+    def test_loading_builders(self):
+        BuildHistory.populate_builders()
+
+        loaded = False
+        if Builder.objects.count() >= 11:
```
Comment (umeshksingla) [inline app/ports/tests/test_buildhistory.py:15]: I think you can at best check for a non-zero number of objects populated in `BuildHistory` here. We don't know if >=11 is going to be fixed, do we (@mojca)?
I think we should rather test this against a fixed JSON file with the format we expect from the upstream.

File: app/ports/tests/test_buildhistory.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            loaded = True
+        self.assertEquals(loaded, True)
+
+    def test_fetch_and_filter_builhistory(self):
```
Comment (umeshksingla) [inline app/ports/tests/test_buildhistory.py]: typo in `...builhistory`

File: app/ports/tests/test_buildhistory.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+from MacPorts.config import TEST_PORTINDEX_JSON, BUILDS_FETCHED_COUNT
+
+
+class TestDependencies(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        Port.load(TEST_PORTINDEX_JSON)
+
+    def test_loading_builders(self):
+        BuildHistory.populate_builders()
```
Comment (umeshksingla) [inline app/ports/tests/test_buildhistory.py:12]: This is of the least importance. But do you also think that it just seems like a wrong abstraction of having `populate_builders` method in `BuildHistory` model?

File: app/ports/tests/test_buildhistory.py (hunk 1 above)
Comment (umeshksingla) [reply app/ports/tests/test_buildhistory.py:15]: Or is the test intended to test the buildbot API call?

File: app/ports/tests/test_buildhistory.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import os
+
+from django.test import TestCase
+from django.urls import reverse
+
+from ports.models import Dependency, Port, BuildHistory, Builder
+from MacPorts.config import TEST_PORTINDEX_JSON, BUILDS_FETCHED_COUNT
+
+
+class TestDependencies(TestCase):
```
Comment (umeshksingla) [inline app/ports/tests/test_buildhistory.py]: Does this not need be a `TransactionTestCase` with `reset_sequences` enabled?

Comment (arjunsalyan) [reply app/ports/tests/test_buildhistory.py]: This does not need, but for consistency, I would use `TransactionTestCase` here also.

File: app/ports/tests/test_buildhistory.py (hunk 2 above)
Comment (arjunsalyan) [reply app/ports/tests/test_buildhistory.py]: Thank you.

File: app/ports/tests/test_buildhistory.py (hunk 1 above)
Comment (arjunsalyan) [reply app/ports/tests/test_buildhistory.py:15]: I just thought that we currently have 11 active builders and this number will only increase with time so I used this. But I am not sure.

File: app/ports/tests/test_buildhistory.py (hunk 3 above)
Comment (arjunsalyan) [reply app/ports/tests/test_buildhistory.py:12]: Hmm. It should have been `Builder.populate_builders` in the `Builder` model.

Comment (arjunsalyan) [reply app/ports/tests/test_buildhistory.py:12]: Shall I change this in this PR only?

**Comment by umeshksingla** [comment]
No, let’s keep it for later. It’s a model change. I just wanted to know
your opinion as well.

File: app/ports/tests/test_buildhistory.py (hunk 4 above)
Comment (umeshksingla) [reply app/ports/tests/test_buildhistory.py]: No, I didn't mean for consistency across the project but really the tests itself.

File: app/ports/tests/test_buildhistory.py (hunk 3 above)
Comment (umeshksingla) [reply app/ports/tests/test_buildhistory.py:12]: No, let’s keep it for later. It’s a model change. I just wanted to know
your opinion as well.

File: app/ports/tests/test_buildhistory.py (hunk 1 above)
Comment (umeshksingla) [reply app/ports/tests/test_buildhistory.py:15]: Let's ask @mojca.

**Comment by arjunsalyan** [comment]
Some improvements were done to how we handle build history and related tests were added. This PR can be closed now.

**Pull request closed without merging**